src/Card.java

```diff
@@ -1,60 +1,44 @@
+import java.util.Arrays;
 import java.util.HashMap;
+import java.util.List;
+import java.util.stream.IntStream;
 
 public class Card {
 
-    String cardCode;
+    private final String cardCode;
+    private final String cardColor;
+    private final String cardValue;
+    private Integer intCardValue;
+    private final HashMap<String, Integer> alphabeticCardValue;
+    private final List<String> colors = Arrays.asList("S","C","D","H");
+    private final List<String> letterValues = Arrays.asList("J","Q","K","A");
 
-    public Card(String cardCode) throws IllegalArgumentException {
-        this.cardCode = cardCode;
+    public Card(String cardCode) {
+        this.cardCode = cardCode.toUpperCase();
+        cardColor = this.cardCode.substring(0, 1);
+        cardValue = this.cardCode.substring(1);
+        alphabeticCardValue = new HashMap<>();
+        IntStream.range(0,4).forEach(i -> alphabeticCardValue.put(letterValues.get(i),i+11));
```

Comment:
Azt altalaban probald elkerulni, hogy list-be, map-be stb. a stream foreach pakolja be az elemeket. Ha rendeltetesszeruen hasznalod a stream-eket, erre altalaban nincs szukseg:
```
alphabeticCardValue = IntStream.range(0,4)
    .boxed()
    .collect(toMap(letterValues::get, i -> i+11))
```